### ISSUE TITLE
error: add-on nptl has no configure fragment or sysdeps tree

### ISSUE THREAD

**Issue by nephaste** [crosstool-ng/crosstool-ng#1831]
- crosstool-NG version 1.25.0.40_994767d
- Host : Ubuntu 22.04 x64 

[conf+buildlog.zip](https://github.com/crosstool-ng/crosstool-ng/files/9758371/conf%2Bbuildlog.zip)

getting mad with an error

using Kernel 5.10.60 / GCC 12.1 / glibc 2.21 (from GNU source)

i am stuck on compilation with  , i have no clue how i can fix this (i didnt enabled Multilib )
may be a flag in ct-ng to modify, i have no clue

checking for g++... g++
[CFG  ]      configure: WARNING: using cross tools not prefixed with host triplet
[CFG  ]      checking whether we are using the GNU C++ compiler... yes
[CFG  ]      checking whether g++ accepts -g... yes
[CFG  ]      checking for x86_64-QNAP_FW501_GCC12-linux-gnu-readelf... x86_64-QNAP_FW501_GCC12-linux-gnu-readelf
[CFG  ]      checking for sysdeps preconfigure fragments... aarch64 alpha arm hppa i386 m68k microblaze mips nios2 powerpc s390 sh sparc tile x86_64 checking whether x86_64-QNAP_FW501_GCC12-linux-gnu-gcc  -g -O2 -fcommon -U_FORTIFY_SOURCE          -Wno-missing-attributes -Wno-array-bounds -Wno-array-parameter -Wno-stringop-overflow -Wno-maybe-uninitialized  compiles in -mx32 mode by default... no
[CFG  ]      
[CFG  ]      configure: running configure fragment for add-on libidn
[CFG  ]      checking for assembler and linker STT_GNU_IFUNC support... yes
[CFG  ]      checking whether .text pseudo-op must be used... yes
[CFG  ]      checking sysdep dirs... sysdeps/unix/sysv/linux/x86_64/64 sysdeps/unix/sysv/linux/x86_64 sysdeps/unix/sysv/linux/x86 sysdeps/unix/sysv/linux/wordsize-64 sysdeps/x86_64/nptl sysdeps/unix/sysv/linux sysdeps/nptl sysdeps/pthread sysdeps/gnu sysdeps/unix/inet sysdeps/unix/sysv sysdeps/unix/x86_64 sysdeps/unix sysdeps/posix sysdeps/x86_64/64 sysdeps/x86_64/fpu/multiarch sysdeps/x86_64/fpu sysdeps/x86/fpu sysdeps/x86_64/multiarch sysdeps/x86_64 sysdeps/x86 sysdeps/ieee754/ldbl-96 sysdeps/ieee754/dbl-64/wordsize-64 sysdeps/ieee754/dbl-64 sysdeps/ieee754/flt-32 sysdeps/wordsize-64 sysdeps/ieee754 sysdeps/generic
[ERROR]      configure: error: add-on nptl has no configure fragment or sysdeps tree
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: glibc_backend_once[scripts/build/libc/glibc.sh@205]
[ERROR]  >>        called from: CT_IterateMultilibs[scripts/functions@1607]
[ERROR]  >>        called from: glibc_main[scripts/build/libc/glibc.sh@26]
[ERROR]  >>        called from: do_libc_main[scripts/build/libc.sh@33]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@697]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >> NOTE: Your configuration includes features marked EXPERIMENTAL.
[ERROR]  >> Before submitting a bug report, try to reproduce it without enabling
[ERROR]  >> any experimental features. Otherwise, you'll need to debug it
[ERROR]  >> and present an explanation why it is a bug in crosstool-NG - or
[ERROR]  >> preferably, a fix.
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]  
[ERROR]  (elapsed: 21:08.32)

any help is welcome ... a week, i am trying to fix it ...

**Comment by nephaste** [comment]
seems working when selecting libc 2.19  ... i wonder to  know how fix glibc 2.21 to work with Crosstool-ng

**Comment by cpackham** [comment]
Hmm, not sure we're gonna be much help here. glibc-2.21 was obsoleted a while ago and has been dropped from ct-ng master. I tried the config you posted (adjusted not to use the custom versions) with the last version of ct-ng that had support for glibc-2.21 and it built fine.

I'm not sure what patches you have applied to your local glibc-2.21 source (ct-ng did have a few but none obviously related to configuring ntpl).

**Comment by nephaste** [comment]
yes .. my System (Qnap NAS) runs under 2.21 libc ... i just pick up the one in GNU Source website (also tried the one from github) , i didnt applied any patches , used nativ sources.. and cannot found any information on this problem browsing internet

each time i am stuck on this nptl error...

i changed to 2.19 from native source ct-bg tree (not the custom config) and it works ...
but getting many linking libs issue (undefined reference to ... ) need to force each libs in autoconf LIBS flag (-llibs) when there are a libs dependency required for the source  .. i was hoping using 2.21 could solved this issue as well...

**Comment by nephaste** [comment]
May be a clue, on 2.20 changelog

* The NPTL implementation of POSIX pthreads is no longer an "add-on".
  On configurations that support it (all Linux configurations), it's now
  used regardless of the --enable-add-ons switch to configure.  It is no
  longer possible to build such configurations without pthreads support.
